org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/testexecution/TestStatusMapperTest.xtend

```diff
@@ -223,4 +225,66 @@ class TestStatusMapperTest extends AbstractPersistenceTest {
 		assertThat(actualStatus).isEqualTo(TestStatus.FAILED)
 	}
 
+	@Test
+	def void getAllInitiallyReturnsEmptyArray() {
+		// given + when
+		val actualStatuses = statusMapperUnderTest.all
+
+		// then
+		assertThat(actualStatuses).isEmpty()
+	}
+
+	@Test
+	def void getAllReturnsStatusForAllRunningTests() {
+		// given
+		val failedTestPath = '/path/to/failedTest.tcl'
+		val failedProcess = mockedTerminatedProcess(EXIT_FAILURE)
+
+		val successfulTestPath = '/path/to/successfulTest.tcl'
+		val successfulProcess = mockedTerminatedProcess(EXIT_SUCCESS)
+
+		val runningTestPath = '/path/to/runningTest.tcl'
+		val runningProcess = mockedRunningProcess()
+
+		statusMapperUnderTest.addTestRun(failedTestPath, failedProcess)
+		statusMapperUnderTest.addTestRun(successfulTestPath, successfulProcess)
+		statusMapperUnderTest.addTestRun(runningTestPath, runningProcess)
+
+		// when
+		val actualStatuses = statusMapperUnderTest.all
+
+		// then
+		actualStatuses => [
```

Comment:
maybe a 'contains' might be easier to read?

Reply:
… went with `containsOnly`. Also, I added the `@EqualsHashCode` active annotation to `TestStatusInfo` (which kinda should have been there from the get-go, anyways) to make this work.